src/proposals/mips/mip-m29/mip-m29.sol

```diff
@@ -0,0 +1,303 @@
+            true
+        );
+
+        _pushHybridAction(
+            addresses.getAddress("mxcDOT"),
+            abi.encodeWithSignature(
+                "_setReserveFactor(uint256)",
+                NEW_MXC_DOT_RESERVE_FACTOR
+            ),
+            "Set reserve factor for mxcDOT to updated reserve factor",
+            true
+        );
+
+        _pushHybridAction(
+            addresses.getAddress("mETHwh"),
+            abi.encodeWithSignature(
+                "_setReserveFactor(uint256)",
+                NEW_M_ETHWH_RESERVE_FACTOR
+            ),
+            "Set reserve factor for mETHwh to updated reserve factor",
+            true
+        );
+
+        _pushHybridAction(
+            addresses.getAddress("UNITROLLER"),
+            abi.encodeWithSignature(
+                "_setCollateralFactor(address,uint256)",
+                addresses.getAddress("mETHwh"),
+                NEW_M_ETHWH_COLLATERAL_FACTOR
+            ),
+            "Set collateral factor of mETHwh",
+            true
+        );
+
+        _pushHybridAction(
+            addresses.getAddress("UNITROLLER"),
+            abi.encodeWithSignature(
+                "_setCollateralFactor(address,uint256)",
+                addresses.getAddress("mUSDCwh"),
+                NEW_M_USDCWH_COLLATERAL_FACTOR
+            ),
+            "Set collateral factor of mUSDCwh",
+            true
+        );
+
+        _pushHybridAction(
+            addresses.getAddress("mxcUSDC"),
+            abi.encodeWithSignature(
+                "_setInterestRateModel(address)",
+                addresses.getAddress("JUMP_RATE_IRM_mxcUSDC")
+            ),
+            "Set interest rate model for mxcUSDC to updated rate model",
+            true
+        );
+
+        _pushHybridAction(
+            addresses.getAddress("mxcUSDT"),
+            abi.encodeWithSignature(
+                "_setInterestRateModel(address)",
+                addresses.getAddress("JUMP_RATE_IRM_mxcUSDT")
+            ),
+            "Set interest rate model for mxcUSDT to updated rate model",
+            true
+        );
+
+        _pushHybridAction(
+            addresses.getAddress("mFRAX"),
+            abi.encodeWithSignature(
+                "_setInterestRateModel(address)",
+                addresses.getAddress("JUMP_RATE_IRM_mFRAX")
+            ),
+            "Set interest rate model for mFRAX to updated rate model",
+            true
+        );
+
+        // Adding transferFrom actions
+        _pushHybridAction(
+            addresses.getAddress("WELL"),
+            abi.encodeWithSignature(
+                "transferFrom(address,address,uint256)",
+                0x6972f25AB3FC425EaF719721f0EBD1Cdb58eE451,
+                0x7793E08Eb4525309C46C9BA394cE33361A167ba4,
+                6778847000000000000000000
+            ),
+            "Transfer 6778847 WELL from 0x6972f25AB3FC425EaF719721f0EBD1Cdb58eE451 to 0x7793E08Eb4525309C46C9BA394cE33361A167ba4",
```

Comment:
Verified that this matches the same call made in April: `Well.transferFrom(0x6972f25AB3FC425EaF719721f0EBD1Cdb58eE451, 0x7793E08Eb4525309C46C9BA394cE33361A167ba4, 6778847000000000000000000)`